packages/back-end/src/models/ExperimentModel.ts

```diff
@@ -802,4 +805,54 @@ export const logExperimentDeleted = async (
   return emittedEvent.id;
 };
 
-// endregion Events
+const isValidVisualExperiment = (
+  e: Partial<VisualExperiment>
+): e is VisualExperiment => !!e.experiment && !!e.visualChangeset;
+
+export const getAllVisualExperiments = async (
+  organization: string
+): Promise<Array<VisualExperiment>> => {
+  const visualChangesets = await findVisualChangesets(organization);
+
+  if (!visualChangesets.length) return [];
+
+  const visualChangesByExperimentId = visualChangesets.reduce(
+    (acc: Record<string, Array<VisualChange>>, c) => {
+      if (!acc[c.experiment]) acc[c.experiment] = [];
+      acc[c.experiment] = acc[c.experiment].concat(c.visualChanges);
+      return acc;
+    },
+    {}
+  );
+
+  const experiments = (
+    await findExperiments({
+      id: {
+        $in: visualChangesets.map((changeset) => changeset.experiment),
+      },
+      organization,
+      archived: false,
+    })
+  )
+    // exclude experiments that are stopped and don't have a released variation
+    .filter((e) => e.status !== "stopped" || e.releasedVariationId !== null)
+    // If the experiment is stopped and the released variation doesn’t have any visual changes, it should be excluded
+    .filter(
+      (e) =>
+        e.status !== "stopped" ||
+        visualChangesByExperimentId[e.id].some(
+          (vc) =>
+            vc.variation === e.releasedVariationId &&
+            (!!vc.css || !!vc.domMutations.length)
+        )
+    );
```

Comment:
Do we need 2 separate `.filter()` iterations? Wondering if it might make sense to have a single `.filter` with early returns to avoid an extra iteration. 

It doesn't seem necessary to check for stopped status multiple times. It looks like all experiments that are not stopped are included so that could be one of the earlier statements, e.g. `if (e.status !== 'stopped') return true`. And a similar check for `releasedVariationId`. Then you could do the `.some()` check by itself on the final line. I think that could help with the legibility as well.

Reply:
I did it this way to make it easier to decipher the business logic for us (or new devs) in the future. I can make the optimizations you've asked - I think it will be less easier to grok however

Reply:
Updated

Reply:
I see the change and I think it's easier to read, and also does one less iteration 👍